src/test/java/seedu/address/logic/commands/UnmatchCommandTest.java

```diff
@@ -0,0 +1,196 @@
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_JOB;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_PERSON;
+import static seedu.address.testutil.TypicalIndexes.INDEX_OUT_OF_BOUNDS;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_JOB;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_PERSON;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.Messages;
+import seedu.address.model.AddressBook;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.company.Company;
+import seedu.address.model.job.Job;
+import seedu.address.model.person.Person;
+import seedu.address.testutil.PersonBuilder;
+import seedu.address.testutil.TypicalCompanies;
+import seedu.address.testutil.TypicalJobs;
+import seedu.address.testutil.TypicalPersons;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for UnmatchCommand.
+ */
+public class UnmatchCommandTest {
+
+    private Model model;
+    private Model expectedModel;
+
+    @BeforeEach
+    public void setUp() {
+        model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+        expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+    }
+
+    @Test
+    public void constructor_nullIndex_throwsNullPointerException() {
+        Index validIndex = Index.fromOneBased(1);
+        assertThrows(NullPointerException.class, () -> new UnmatchCommand(null, null));
+        assertThrows(NullPointerException.class, () -> new UnmatchCommand(null, validIndex));
+        assertThrows(NullPointerException.class, () -> new UnmatchCommand(validIndex, null));
+    }
+
+    @Test
+    public void execute_validContactAndJob_unmatchSuccessful() throws Exception {
+        Index contactIndex = INDEX_FIRST_PERSON;
+        Index jobIndex = INDEX_FIRST_JOB;
+        // First, set up the model with an existing match
+        MatchCommand initialMatchCommand = new MatchCommand(contactIndex, jobIndex);
+        initialMatchCommand.execute(model);
+
+        UnmatchCommand unmatchCommand = new UnmatchCommand(contactIndex, jobIndex);
+
+        Person contactToUnmatch = model.getFilteredPersonList().get(contactIndex.getZeroBased());
+        Job jobToUnmatch = model.getFilteredJobList().get(jobIndex.getZeroBased());
+
+        // Ensure the contact is currently matched to the job
+        assert contactToUnmatch.hasMatched(jobToUnmatch.getIdentifier());
+        // Execute the unmatch command to update the model's state
+        CommandResult commandResult = unmatchCommand.execute(model);
```

Comment:
It seems that you are manually executing unmatchCommand and checking the changes in model, however, aren't `assertCommandSuccess()` already done that for you?

Reply:
I think `assertCommandSuccess()` should be used as well.